Makefile

```diff
@@ -161,6 +161,15 @@ e2e-tests-compile:
 # Compile e2e tests for each platform.
 build.code.platform: e2e-tests-compile
 
+# NOTE(hasheddan): the build submodule currently overrides XDG_CACHE_HOME in
+# order to force the Helm 3 to use the .work/helm directory. This causes Go on
+# Linux machines to use that directory as the build cache as well. We should
+# adjust this behavior in the build submodule because it is also causing Linux
+# users to duplicate their build cache, but for now we just make it easier to
+# identify its location in CI so that we cache between builds.
```

Comment:
It is possible that we do want to replicate the build cache across projects, and this is actually working for speeding up builds locally because we are pointing at the same location every time. However, we are not caching this between builds in CI because `go env GOCACHE` outside of our `make` context is not giving us the same path as inside of our `make` context.